RetailCoder.VBE/UI/Refactorings/ReorderParameters/ReorderParametersDialog.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Windows.Forms;
+using Rubberduck.Parsing.Symbols;
+
+namespace Rubberduck.UI.Refactorings.ReorderParameters
+{
+    public partial class ReorderParametersDialog : Form, IReorderParametersView
+    {
+        public List<Parameter> Parameters { get; set; }
+        public Parameter SelectedItem { get; set; }
+
+        public ReorderParametersDialog()
+        {
+            Parameters = new List<Parameter>();
+            InitializeComponent();
+
+            MethodParametersGrid.SelectionChanged += MethodParametersGrid_SelectionChanged;
+        }
+
+        private void MethodParametersGrid_SelectionChanged(object sender, EventArgs e)
+        {
+            DataGridView sentVal = sender as DataGridView;
+            SelectedItem = Parameters.First(item => item.Name == (string)sentVal.CurrentCell.Value);
+        }
+
+        public void InitializeParameterGrid()
+        {
+            MethodParametersGrid.AutoGenerateColumns = false;
+            MethodParametersGrid.Columns.Clear();
+            MethodParametersGrid.DataSource = Parameters;
+            MethodParametersGrid.AlternatingRowsDefaultCellStyle.BackColor = Color.Lavender;
+            MethodParametersGrid.MultiSelect = false;
+
+            var paramNameColumn = new DataGridViewTextBoxColumn();
+            paramNameColumn.Name = "Name";
+            paramNameColumn.DataPropertyName = "Name";
+            paramNameColumn.HeaderText = "Name";
+            paramNameColumn.ReadOnly = true;
+            paramNameColumn.Width = 262;    // fits nice
+
+            MethodParametersGrid.Columns.Add(paramNameColumn);
+            SelectedItem = Parameters[0];
+        }
+
+        private void OkButtonClick(object sender, EventArgs e)
+        {
+            OnOkButtonClicked();
+        }
+
+        public event EventHandler CancelButtonClicked;
+        public void OnCancelButtonClicked()
+        {
+            Hide();
+        }
+
+        public event EventHandler OkButtonClicked;
+        public void OnOkButtonClicked()
+        {
+            var handler = OkButtonClicked;
+            if (handler != null)
+            {
+                handler(this, EventArgs.Empty);
+            }
+        }
+
+        private Declaration _target;
+        public Declaration Target
+        {
+            get { return _target; }
+            set
+            {
+                _target = value;
+                if (_target == null)
+                {
+                    return;
+                }
+            }
```

Comment:
I don't quite understand what you're doing here.  You're returning at the same time the method would return anyway. Did you mean to check the `value` for null prior to setting?? If so, an argument exception should be thrown, no?